src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,278 @@
+          icdm_attributes.ActiveModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"ActiveModeDuration: {active_mode_duration}")
+
+        self.step("1b")
+
+        # Subscription to ActiveChanged
+        event = brb_info_cluster.Events.ActiveChanged
+        self.q = queue.Queue()
+        urgent = 1
+        cb = SimpleEventCallback("ActiveChanged", event.cluster_id, event.event_id, self.q)
+        subscription = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=[(kICDBridgedEndpointId, event, urgent)], reportInterval=[1, 3])
+        subscription.SetEventUpdateCallback(callback=cb)
+
+        stay_active_duration = 1000
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration}ms)")
+        self._send_keep_active_command(stay_active_duration)
+
+        logging.info("Waiting for ActiveChanged from DUT...")
+        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+
+        asserts.assert_true(promised_active_duration >= stay_active_duration, "PromisedActiveDuration < StayActiveDuration")
+
+        self.step("2")
+
+        stay_active_duration = 1500
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration}ms)")
+        self._send_keep_active_command(stay_active_duration)
+
+        logging.info("Waiting for ActiveChanged from DUT...")
+        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+
+        # wait for active time duration
+        time.sleep(max(stay_active_duration/1000, active_mode_duration))
+        # ICD now should be in idle mode
+
+        # sends 3x keep active commands
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
+        self._send_keep_active_command(stay_active_duration)
+        time.sleep(100)
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
+        self._send_keep_active_command(stay_active_duration)
+        time.sleep(100)
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
+        self._send_keep_active_command(stay_active_duration)
+        time.sleep(100)
+
+        logging.info("Waiting for ActiveChanged from DUT...")
+        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+
+        asserts.assert_true(self.q.qSize() == 0, "More than one event received from DUT")
+
+        self.step("3")
+
+        stay_active_duration = 10000
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
+        self._send_keep_active_command(stay_active_duration)
+
+        ## halt the ICD process
+        self.app_process.send_signal(signal.SIGSTOP.value)
+
+        if not self.is_ci:
+            logging.info("Waiting for 60 minutes")
```

Comment:
To do this, you'll need to adjust the test timeout. Right now it's on the order of minutes. You need WAY more than that. Actually, are you not already seeing timeouts on this test with the sleeps above?

Reply:
done